tooling/strace_analysis.sh

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+# Before executing this script, strace output files need to be generated
+# $1 is path of strace output folder
+# $2 is path of temurin-build folder, for exmaple: /home/user/Documents/temurin-build"
+# $3 is javaHome
+# $4 is classpath
+# $5 is sbomJson
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "$SCRIPT_DIR/../sbin/common/sbom.sh"
+
+declare -g javaHome
+declare -g classpath
+declare -g sbomJson
+
+# Arrays to store different types of strace output, to treat them different
+declare -g usrLocalFiles=()
+declare -g otherFiles=()
+
+# Arrays for package and non-package dependencies
+declare -g pkgs=()
+declare -g npkgs=()
+
+# Array to store packages, to make sure no duplicates are added to Sbom
+declare -g uniqueVersions=()
+
+declare -g isBinSymLink=false
+declare -g isLibSymLink=false
+declare -g isSbinSymLink=false
+
+# ignore-patterns for strace files
+declare -g ignores=(
+  "\.java$"
+  "\.d$"
+  "\.o$"
+  "\.d\.targets$"
+  "^$2"
+  "\+\+\+"
+  "\-\-\-"
+  "^/dev/"
+  "^/proc/"
+  "^/sys/"
+  "^/tmp/"
+)
+
+checkArguments() {
+  if [ -z "$1" ]; then
+      echo "strace output folder as param is missing!"
```

Comment:
Whitespace inconsistency through this file.

Reply:
I reformated the whole file now and pushed it.